File: leetcode/455.assign-cookies/go/findcontentchildren.go
package assigncookie

import "slices"

func findContentChildren(g []int, s []int) int {
	slices.Sort(g)
	slices.Sort(s)
	res := 0
	j := 0

	for i := range s {
		if j >= len(g) {
			break
		}

		if s[i] >= g[j] {
			j++
			res++
		}
	}

	return res
}

File: leetcode/455.assign-cookies/cpp/findcontentchildren.cpp
#include <vector>

class Solution
{
public:
    int findContentChildren(std::vector<int> &g, std::vector<int> &s)
    {
        std::sort(g.begin(), g.end());
        std::sort(s.begin(), s.end());

        int res = 0;
        int j = 0;

        for (auto c : s)
        {
            if (j >= g.size())
                break;

            if (c < g[j])
                continue;

            j++;
            res++;
        }

        return res;
    }
};
